src/index.ts

```diff
@@ -223,6 +333,117 @@ async function getPortalItemInfo(
     return portalInfo;
 }
 
+/**
+ * Gets the set of parameters required for a given print template.
+ * @param itemId The portal item ID of the report item.
+ * @param portalUrl The URL of the ArcGIS Portal instance to use. Defaults to ArcGIS Online: "https://www.arcgis.com".
+ * @param serviceUrl The url to the template service.
+ * @param runToken A token used to authenticate with the service.
+ */
+export async function getMetadata(
+    itemId: string,
+    portalUrl: string,
+    serviceUrl: string,
+    runToken?: string
+): Promise<TemplateMetadata> {
+    // Infer the URL to the reporting service from the item
+    const apiServiceUrl = `${serviceUrl}service`;
+
+    const body = {
+        template: {
+            itemId,
+            portalUrl,
+        },
+    };
+    const headers = {
+        "Content-Type": "application/json",
+    };
+    if (runToken) {
+        headers["Authorization"] = runToken;
+    }
+    const requestOptions = {
+        method: "POST",
+        responseType: "json",
+        body: JSON.stringify(body),
+        headers,
+    };
+
+    try {
+        // Make the metadata request
+        const response = await fetch(
+            `${ensureTrailingSlash(apiServiceUrl)}job/metadata`,
+            requestOptions
+        );
+
+        if (!response.ok) {
+            createError(response.statusText, response.status);
+        }
+
+        // Parse the response
+        const responseJson = (await response.json()) as MetadataResponse;
+
+        const data = responseJson.response;
+        if (!data || !data.parameters) {
+            throw new Error("The print service did not provide any metadata.");
```

Comment:
Let's remove the word "print" from this message. Also on line `400`

Reply:
Done